sysdig/resource_sysdig_secure_cloud_auth_account_test.go

```diff
@@ -441,3 +441,140 @@ func TestAccAWSSecureCloudAccountFCCSPM(t *testing.T) {
+			"sysdig": func() (*schema.Provider, error) {
+				return sysdig.Provider(), nil
+			},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+		resource "sysdig_secure_cloud_auth_account" "gcp-agentless-scanning" {
+			provider_id   = "gcp-agentless-test-%s"
+			provider_type = "PROVIDER_GCP"
+			enabled       = true
+		    feature {
+			  secure_agentless_scanning {
+			    enabled    = true
+			    components = ["COMPONENT_SERVICE_PRINCIPAL/secure-scanning"]
+			  }
+		    }
+			component {
+				type                       = "COMPONENT_SERVICE_PRINCIPAL"
+				instance                   = "secure-scanning"
+				service_principal_metadata = jsonencode({
+					gcp = {
+						workload_identity_federation = {
+							pool_provider_id = "pool_provider_id_value"
+						}
+						email = "email_value"
+					}
+				})
+			}
+		}`, accID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp-agentless-scanning", "provider_type", "PROVIDER_GCP"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp-agentless-scanning", "enabled", "true"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp-agentless-scanning", "feature.0.secure_agentless_scanning.0.enabled", "true"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp-agentless-scanning", "feature.0.secure_agentless_scanning.0.components.0", "COMPONENT_SERVICE_PRINCIPAL/secure-scanning"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp-agentless-scanning", "component.0.type", "COMPONENT_SERVICE_PRINCIPAL"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp-agentless-scanning", "component.0.instance", "secure-scanning"),
+					resource.TestCheckResourceAttr("sysdig_secure_cloud_auth_account.gcp-agentless-scanning", "component.0.service_principal_metadata", "{\"gcp\":{\"email\":\"email_value\",\"workload_identity_federation\":{\"pool_provider_id\":\"pool_provider_id_value\"}}}"),
+				),
+			},
+			{
+				ResourceName:      "sysdig_secure_cloud_auth_account.gcp-agentless-scanning",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestGCPAgentlesScanningOnboardingWithInventory(t *testing.T) {
+	rText := func() string { return acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum) }
+	accountID := rText()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			if v := os.Getenv("SYSDIG_SECURE_API_TOKEN"); v == "" {
+				t.Fatal("SYSDIG_SECURE_API_TOKEN must be set for acceptance tests")
+			}
+		},
+		ProviderFactories: map[string]func() (*schema.Provider, error){
+			"sysdig": func() (*schema.Provider, error) {
+				return sysdig.Provider(), nil
+			},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+		resource "sysdig_secure_cloud_auth_account" "gcp-agentless-scanning" {
```

Comment:
Just curious, did the alignment give you any problem here while running the tests?

Reply:
both on that line and in general,  yes, the current tests strategy is a pain in the ass because it's space-sensitive.
i think that's something worth to investigate; on howto make it more lightweight, if possible, but did not check any options myself.

Reply:
I believe its to do with TF files in general. If there is a func or parser to convert free form formatted string to TF acceptable format, that would make it lightweight.